core/src/main/java/org/apache/iceberg/V2Metadata.java

```diff
@@ -47,7 +47,9 @@ private V2Metadata() {
       ManifestFile.ADDED_ROWS_COUNT.asRequired(),
       ManifestFile.EXISTING_ROWS_COUNT.asRequired(),
       ManifestFile.DELETED_ROWS_COUNT.asRequired(),
-      ManifestFile.PARTITION_SUMMARIES
+      ManifestFile.PARTITION_SUMMARIES,
+      ManifestFile.KEY_METADATA,
```

Comment:
Why does this add key metadata?

Reply:
I am not sure this is right, please correct me. Because I need to keep the schema matching the `IndexedManifestFile.get` method. 

```java
        case 13:
          return wrapped.partitions();
        case 14:
          return wrapped.keyMetadata();
        case 15:
          return wrapped.schemaId();
        default:
          throw new UnsupportedOperationException("Unknown field ordinal: " + pos);
```